### PR TITLE
feat: data generation framework for unhappy paths

### PR DESCRIPTION
**Pull request by palozano**
### Summary by OpenAI

Release Notes:

- New Feature: Added a new command `dgf` to generate data for unhappy paths.
- New Feature: Added a function to submit an extrinsic with raw transaction data.
- Test: Added a bash script to test the maintenance mode of Rococo for t0rn (and possibly t3rn).

> "New commands and tests galore,
> Raw transactions and maintenance mode explore.
> With this PR, we take a leap,
> Improving our code, one feature at a time, deep."

### Summary by OpenAI

Release Notes:

- Bug fix: Fixed an issue where the `creation` function in `client/packages/cli/src/utils/dgf/creation.ts` was not properly handling errors.
- Refactor: Improved error handling and added better logging throughout the codebase.

> "Bugs were squashed, errors tamed,
> Code now runs smoother, less maimed.
> With better logs and handling too,
> This PR brings a better view."

### Summary by OpenAI

Release Notes:

- New Feature: Added `dgf` command to generate data for unhappy paths in the CLI.
- Bug fix: Fixed a bug in the CLI codebase.
- Refactor: Refactored the CLI codebase.
- Test: Added a bash script to test the maintenance mode of Rococo for t0rn (and possibly t3rn).
- Chore: Improved error handling and logging throughout the CLI codebase.

> "Unhappy paths no more,
> With `dgf` we can explore.
> Bugs fixed, code refactored,
> Error handling improved, logs recorded.
> Celebrate with a joyful roar!"

### Summary by OpenAI

Release Notes:

- New Feature: Added a `dgf` command to generate data for unhappy paths, including functions to create random values for different types of data and a module to process SFX files and define an event listener for the relayer.
- Bug fix: Improved error handling and logging throughout the codebase.
- Refactor: Made changes to the codebase to improve its maintainability.

> A new command has arrived,
> To generate data that's not so nice,
> With improved error handling by its side,
> And refactored code that's easy on the eyes.

### Summary by OpenAI

Release Notes:

- New Feature: Added a `dgf` command to generate data for unhappy paths, including functions to create random values for different types of data and a module to process SFX files and define an event listener for the relayer.
- Bug fix: Improved error handling and logging throughout the codebase.
- Refactor: Made changes to improve the codebase.

> "Unhappy paths now have data galore,
> With `dgf` command, we can explore.
> Errors are handled with better care,
> And logging is improved everywhere.
> Refactored code, cleaner than before,
> This PR brings joy, let's all adore!"

### Summary by OpenAI

Release Notes:

- New Feature: Added a `dgf` command to generate data for unhappy paths, including functions to create random values for different types of data and a module to process SFX files and define an event listener for the relayer. Also added a function to submit an extrinsic with raw transaction data and a bash script to test the maintenance mode of Rococo for t0rn (and possibly t3rn).
- Bug fix: Fixed a bug in the codebase.
- Refactor: Refactored the codebase to improve its maintainability and modularity.
- Documentation: Added better logging throughout the codebase.

> A new command has arrived,
> To generate data that's not so nice,
> With improved error handling and logging,
> And a bug fix that's worth applauding.

### Summary by OpenAI

Release Notes:

- New Feature: Added a `dgf` command to generate data for unhappy paths in the CLI tool.
- Bug fix: Fixed a bug in the CLI tool.
- Documentation: Updated the README file for the CLI tool.
- Refactor: Improved maintainability and modularity of the CLI tool.
- Test: Added a bash script to test the maintenance mode of Rococo for t0rn (and possibly t3rn).
- Chore: Improved error handling and logging throughout the codebase.

> "Unhappy paths no more,
> With `dgf` we can explore.
> Bugs fixed, docs updated,
> Codebase improved, refactored, and tested.
> Error handling and logging enhanced,
> Celebrate this PR with a joyful dance!"

### Summary by OpenAI

Release Notes:

- New Feature: Added a `dgf` command to generate data for unhappy paths, including functions to create random values for different types of data and a module to process SFX files and define an event listener for the relayer.
- Bug fix: Improved error handling and logging throughout the codebase.
- Refactor: Refactored the codebase to improve its maintainability and modularity.
- Documentation: Updated the CLI tool README file.

> "Unhappy paths no more,
> With `dgf` we can explore.
> Error handling improved,
> Modularity's groove moved.
> Refactored with care,
> Docs updated to share."

### Summary by OpenAI

Release Notes:

- New Feature: Added a `dgf` command to generate data for unhappy paths, including functions to create random values for different types of data and a module to process SFX files and define an event listener for the relayer. Also added a bash script to test the maintenance mode of Rococo for t0rn (and possibly t3rn).
- Bug fix: Fixed a bug in the codebase.
- Refactor: Refactored the codebase to improve its maintainability and modularity.
- Documentation: Improved error handling and logging throughout the codebase. Updated the CLI tool for interacting with the t3rn circuit.

> Changes abound, new features found,
> Bugs fixed, codebase sound.
> Refactored with care, docs everywhere,
> Celebrate this PR! 🎉

### Summary by OpenAI

Release Notes:

- New Feature: Added a `dgf` command to generate data for unhappy paths, including functions to create random values for different types of data and a module to process SFX files and define an event listener for the relayer. Also added a bash script to test the maintenance mode of Rococo for t0rn (and possibly t3rn).
- Bug fix: Fixed a bug in the codebase.
- Refactor: Refactored the codebase to improve its maintainability and modularity.
- Documentation: Improved error handling and logging throughout the codebase. Added better logging and comments.

> A new command has arrived,
> To generate data that's not so nice,
> With improved error handling by its side,
> And refactoring to make it precise.
> Celebrate this PR with glee,
> For it brings modularity! 🎉

### Summary by OpenAI

Release Notes:

- Refactor: Improved error handling and modularity in `client/packages/cli/src/commands/deploy.ts`.
- Bug fix: Fixed an issue with data races in `server/src/app.ts`.
- Test: Added unit tests for `client/packages/cli/src/utils/logger.ts`.

> "Code improved, bugs fixed,
> Tests added, quality mixed.
> With each PR we strive to be,
> Better than we were previously."

### Release Notes

- Test: Update import statements in test files
- Refactor: Update import statement for `Gateway` in `register/gateway.ts`
- Chore: Skip test cases in `register/__test__/gateway.test.ts`

> "Code changes abound,  
> Logic, security, and more found.  
> With each review,  
> The code quality grew.  
> Let's celebrate with a round!"

### Summary by OpenAI

Refactor: This pull request includes various refactorings and updates to import statements in the client-side codebase of the CLI package. These changes aim to improve code quality, maintainability, and consistency.

> "Code refactored with care,
> Import statements updated everywhere.
> Bugs fixed, tests skipped with ease,
> CLI package now a breeze."

### Summary by OpenAI

Refactor: This pull request includes various refactorings and updates to import statements in the client-side codebase of the CLI package. These changes aim to improve code quality, maintainability, and consistency.

> "Code refactored with care,
> Import statements updated everywhere.
> Bugs fixed, tests skipped,
> Code quality improved, maintainability equipped."

### Summary by OpenAI

**Release Notes**
- Bug fix: Skipped test cases in `gateway.test.ts` due to node-fetch being an ESM module.
- Refactor: Updated import statements in `gateway.ts` and `sfx.test.ts`.
- Chore: Added a configuration file for `tsup` bundler to build the CLI package.

> "Code changes, big and small,
> OpenAI reviewed them all.
> Bugs were fixed, imports updated,
> tsup config added, code consolidated.
> With each PR, the codebase grows,
> And with it, our knowledge flows."

### Summary by OpenAI

Release Notes:

- Refactor: Minor changes to `client/packages/cli/src/cli.ts`.

> "A small tweak here, a small tweak there,
> Our codebase is now beyond compare.
> With every pull request we grow,
> And make our software shine and glow."

### Summary by OpenAI

Release Notes:

- Refactor: Minor code cleanup in `client/packages/cli/src/cli.ts`.

> "A little refactoring here and there,
> makes our codebase clean and fair."

### Summary by OpenAI

Release Notes:

- Refactor: Minor code cleanup in `client/packages/cli/src/cli.ts`.

> "A little refactoring here and there,
> makes our codebase clean and fair."

### Summary by OpenAI

Release Notes:
- Refactor: Minor code cleanup in `client/packages/cli/src/cli.ts`.

> "A little refactoring here and there,
> Makes our codebase clean and fair."

### Summary by OpenAI

Release Notes:

- Refactor: Improved error handling and modularity in `client/packages/cli/src/commands/deploy.ts`.
- Chore: Updated dependencies in `client/packages/cli/package.json`.

> "Bug fixes and improvements galore,
> Deploying code is now a chore no more.
> With better error handling and modularity too,
> This CLI update will surely please you."

### Summary by OpenAI

Release Notes:

- Refactor: Minor code cleanup in `client/packages/cli/src/cli.ts`.

> "A little refactoring here and there,
> makes our codebase clean and fair."

### CODE REVIEW

**Review by ahkohd — APPROVED** [review]
Let's go!

**Review by ahkohd — APPROVED** [review]
Let's go!

File: client/packages/cli/src/commands/dgf.ts [hunk 1]
```diff
@@ -0,0 +1,186 @@
...
+    process.exit(1)
+  }
+
+  spinner.start(
+    colorLogMsg("INFO", "Starting the unhappy path generation process...")
+  )
+
+  try {
+    const provider = new WsProvider(getConfig().circuit.ws)
+    const api = await ApiPromise.create({ provider })
```
Comment (petscheit) [inline client/packages/cli/src/commands/dgf.ts:33]: This should come from SDK instance, as done in the other commands

Comment (ahkohd) [reply client/packages/cli/src/commands/dgf.ts]: Fixed, refer to commit 884d656.

**Pull request merged**